postmark/core.py

```diff
@@ -151,6 +153,18 @@ def _set_custom_headers(self, value):
         else:
             raise TypeError('Custom headers must be a dictionary of key-value pairs')
 
+    def _set_metadata(self, value):
+        '''
+        A special set function to ensure
+        we're setting with a dictionary
+        '''
+        if value is None:
+            setattr(self, '_PMMail__metadata', {})
+        elif isinstance(value, dict):
+            setattr(self, '_PMMail__metadata', value)
+        else:
+            raise TypeError('Metadata must be a dictionary of key-value pairs')
```

Comment:
Can you also check that the metadata dict is a simple `string: string` dictionary? I just tried sending with a more complex set of nested dictionaries and got back an error. We should handle that in here, in my opinion. Thanks!